File: cblmapper/src/main/java/io/sportner/cblmapper/exceptions/UnhandledTypeException.java
package io.sportner.cblmapper.exceptions;

public class UnhandledTypeException extends CBLMapperClassException {
    String mMessage;

    public UnhandledTypeException(Class unhandledClass) {
        super(unhandledClass);
        this.mMessage = String.format("Class '%s' has no type adapter", unhandledClass.getName());
    }

    public String getMessage() {
        return this.mMessage;
    }
}

File: cblmapper/src/main/java/io/sportner/cblmapper/exceptions/UnhandledTypeException.kt
package io.sportner.cblmapper.exceptions

import java.lang.reflect.Type
import kotlin.reflect.KClass

class UnhandledTypeException(unhandledType: Type) : CBLMapperClassException(unhandledType, String.format("Class '%s' has no type adapter", unhandledType.toString())) {
    constructor(unhandledKClass: KClass<*>): this(unhandledKClass.java)
    constructor(unhandledClass: Class<*>): this(unhandledClass as Type)
}
